File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/LongHexTextFormatterFilter.java
package com.cedarsoft.commons.javafx;

import java.util.function.Predicate;

import javax.annotation.Nonnull;

/**
 * Processes text formatter changes. Ensures only valid longs are added
 */
class LongHexTextFormatterFilter extends NumberTextFormatterFilter {

  LongHexTextFormatterFilter(@Nonnull NumberStringConverterForLongHex converter, @Nonnull Predicate<Long> filter) {
    super(converter, number -> filter.test(number.longValue()), 0);
  }

}

File: commons/javafx/src/main/kotlin/it/neckar/open/javafx/LongHexTextFormatterFilter.kt
package it.neckar.open.javafx

import java.util.function.Predicate

/**
 * Processes text formatter changes. Ensures only valid longs are added
 */
class LongHexTextFormatterFilter(
  converter: NumberStringConverterForLongHex,
  filter: Predicate<Long>,
) : NumberTextFormatterFilter(
  converter = converter,
  filter = Predicate { number: Number -> filter.test(number.toLong()) },
  emptyValue = 0
)
